### ISSUE TITLE
JavaScript code being appended to every url served from cdn.jsdelivr.net (hacked?)

### ISSUE THREAD

**Issue by coolreader18** [jsdelivr/jsdelivr#18070]
[here](https://cdn.jsdeliver.net/npm/primer@10.4.0/build/build.css), [here](https://cdn.jsdeliver.net/npm/primer@10.4.0/build/), [here](https://cdn.jsdeliver.net/npm/primer@10.4.0), and on the [error pages](https://cdn.jsdeliver.net/npm/). Has jsDelivr been hacked, as [looking it up](https://www.google.com/search?q=freecontent.date) yields "GET RID OF THE FREECONTENT.DATE VIRUS", which I'm sure are clickbait/template sites, but it's still worrying.

[Wayback machine snapshot](https://web.archive.org/web/20180508234026/https://cdn.jsdeliver.net/npm/primer@10.4.0/build/)

**Comment by coolreader18** [comment]
@MartinKolarik @jimaek

**Comment by jimaek** [comment]
Indeed, we do not control that domain. I am not sure what can be done in this case. I doubt the domain registrar would do anything. But I will try to get in contact with them.

**Comment by jimaek** [comment]
They are hosted on https://www.singlehop.com/
This way their response:

"Thank you for your email. While we appreciate the seriousness of your complaint, we are unable to take action at this time. As a courtesy to you, we’ve forwarded your complaint to our customer. "

**Comment by thezedwards** [comment]
Howdy ya'll -- it unfortunately seems like this domain is active again. I believe the group who has it is doing very targeted attacks. It's different than before but if you go to the homepage or try to request specific packages, you'll get injections and redirects.

**Comment by MartinKolarik** [comment]
@thezedwards can you provide a sample link? It doesn't seem to be working for me.

**Comment by thezedwards** [comment]
It has orchestration on it to redirect traffic to a landing page sometimes - if you hit the page directly it will sometimes just push you to a WW1 landing page subdomain w/o any exploits. 

You can see some malicious scans @ https://any.run/report/eb7a1cd86aecc132b17bb87e2cb1e4960bec8e07dd443424dd790360ebe0e03c/2b54d486-da1f-43db-9ed5-8fb74703edfd

**Comment by thezedwards** [comment]
They've got a ton of subdomains mapped that you can view @ https://securitytrails.com/list/apex_domain/jsdeliver.net

IMO this gives you a sense of the scale..

**Comment by jimaek** [comment]
I reported the abuse to the new registar. Lets see if that solves the problem

**Comment by jimaek** [comment]
>Thank you for submitting your abuse complaint. We will investigate the matter, and take action where we deem appropriate.

>Please understand that due to the volume of complaints we receive, you may not receive any further response from us, unless we require more information from you.

**Comment by myrriad** [comment]
According to github search, there are currently [245 instances](https://github.com/search?l=HTML&q=cdn.jsdeliver.net&type=Code) of the unsafe jsdelivr link. Is there any way to notify these people that they are running malicious code?

**Comment by jimaek** [comment]
I dont think its active, all those links 404. But I will try to contact the registrar again
